pkg/gardenlet/controller/shoot/shoot/reconciler_reconcile.go

```diff
@@ -606,6 +606,16 @@ func (r *Reconciler) runReconcileShootFlow(ctx context.Context, o *operation.Ope
 			Fn:           flow.TaskFn(botanist.DeployWorker).RetryUntilTimeout(defaultInterval, defaultTimeout),
 			Dependencies: flow.NewTaskIDs(deployCloudProviderSecret, deployReferencedResources, waitUntilInfrastructureReady, initializeShootClients, waitUntilOperatingSystemConfigReady, waitUntilNetworkIsReady, createNewServiceAccountSecrets),
 		})
+		waitUntilWorkerStatusUpdate = g.Add(flow.Task{
+			Name:         "Waiting until worker resource status is updated with latest machine deployments",
+			Fn:           botanist.Shoot.Components.Extensions.Worker.WaitUntilWorkerStatusMachineDeploymentsUpdated,
```

Comment:
Shouldn't we skip this wait if `skipReadiness==true` similar to https://github.com/gardener/gardener/blob/14b7b281d57cfd42529151c448f8c0d84ba4f68b/pkg/gardenlet/controller/shoot/shoot/reconciler_reconcile.go#L626
```suggestion
			Fn:           flow.TaksFn(botanist.Shoot.Components.Extensions.Worker.WaitUntilWorkerStatusMachineDeploymentsUpdated).SkipIf(skipReadiness),
```

Reply:
If this step is skipped, we won't be populating `w.machineDeployments`; thus, the `--nodes` flag in the cluster autoscaler won't be present [See [this](https://github.com/gardener/gardener/blob/5abed1bfedd6f2188e5dcec9af161d1b5c286236/pkg/operation/botanist/component/clusterautoscaler/cluster_autoscaler.go#L122) and [this](https://github.com/gardener/gardener/blob/5abed1bfedd6f2188e5dcec9af161d1b5c286236/pkg/operation/botanist/component/clusterautoscaler/cluster_autoscaler.go#L416)]. This will result in Cluster autoscaler being in a `CrashLoopBackOff` phase. It will eventually be successfully deployed once `skipReadiness==false` (which might happen in the next reconciliation itself) but till then we won't have a running CA.

Reply:
Yep, I was mainly asking because this was the old behaviour when the `Wait()` for the worker is skipped? We were populating the `w.machineDeployments` in the `Wait()` here: https://github.com/gardener/gardener/blob/b45935926e21941a6a4bae72db1045af69bff35e/pkg/operation/botanist/component/extensions/worker/worker.go#L285-L301

@timuthy,  iiuc the requirements of https://github.com/gardener/gardener/issues/7224, this step should be skipped, right?

Reply:
Good point @plkokanov, but I think we shouldn't skip the `WaitUntilWorkerStatusMachineDeploy` step when `skipReadiness` is requested.
`skipReadiness` is usually enabled in case of an AZ outage or similar and there it becomes especially important to set the desired/intended `MachineDeployment`s in the CA config. Also, we literally don't wait for the worker's readiness, only for the `MachineDeployment`s to be applied.
Do you have anything else in mind that I didn't consider?

Reply:
Thanks for the explanation @timuthy and @rishabh-11!
> Also, we literally don't wait for the worker's readiness, only for the MachineDeployments to be applied. 

I see, then everything looks fine as it is from my pov.